mixnet/client/src/receiver.rs

```diff
@@ -7,21 +7,30 @@ use nym_sphinx::{
     message::{NymMessage, PaddedMessage},
     Payload,
 };
-use tokio::net::TcpStream;
+use tokio::{net::TcpStream, sync::Mutex};
 
 use crate::MixnetClientError;
 
 // Receiver accepts TCP connections to receive incoming payloads from the Mixnet.
-pub struct Receiver;
+pub struct Receiver {
+    node_address: SocketAddr,
+}
 
 impl Receiver {
+    pub fn new(node_address: SocketAddr) -> Self {
+        Self { node_address }
+    }
+
     pub async fn run(
-        node_address: SocketAddr,
+        &self,
     ) -> Result<
         impl Stream<Item = Result<Vec<u8>, MixnetClientError>> + Send + 'static,
         MixnetClientError,
     > {
-        let Ok(socket) = TcpStream::connect(node_address).await else {
+        let Ok(socket) = TcpStream::connect(self.node_address)
+            .await
+            .map(|s| Arc::new(Mutex::new(s)))
```

Comment:
maybe we can remove this mutex now?

Reply:
In logic, we can remove the mutex, but we still need a mutex to provide internal mutability.

Reply:
Ah, sorry I don't get. maybe to cover the case when multi threads access the fragment stream?

Reply:
Ah, it is because `Body::read(&mut *mu)` requires a mutable reference, and if we remove the Mutex, then we cannot get the mutable reference from `Arc`. We have another `Mutex` on `HashMap`, but we still have to wrap `TcpStream` into an `Arc`, `TcpStream` is not clonable, and we will use this mutable reference in an async closure which requires `'static` lifetime.

Reply:
Aha. but how about removing both Arc and Mutex from this `receiver.rs`? because we're not gonna use the connection map in this code. I mean reverting this code entirely.

Reply:
Aha, yeah, we can totally remove Arc and Mutex, thanks, I forget we are not going to use conn map in this file.

Reply:
Aha, yeah, we can totally remove Arc and Mutex, thanks, I forget we are not going to use conn map in this file.